sflow/agent.go

```diff
@@ -91,19 +95,166 @@ func (sfa *Agent) feedFlowTable() {
 				// iterate over a set of Packets as a sample contains multiple
 				// records each generating Packets.
 				sfa.FlowTable.FeedWithSFlowSample(&sample, bpf)
 			}
+
+			// SFlow Counter Samples
+			Countersamples := sflowPacket.CounterSamples
+
+			var gen layers.SFlowGenericInterfaceCounters
+			var ovsdp layers.SFlowOVSDPCounters
+			var app layers.SFlowAppresourcesCounters
+			var vlan layers.SFlowVLANCounters
+
+			for _, sample := range sflowPacket.CounterSamples {
+				records := sample.GetRecords()
+
+				maxuint64 := func(key uint64) uint64 {
+					if key == 18446744073709551615 {
+						key = 0
+					}
+					return key
+				}
+				maxuint32 := func(key uint32) uint32 {
+					if key == 4294967295 {
+						key = 0
+					}
+					return key
+				}
+
+				for _, record := range records {
+					switch record.(type) {
+					case layers.SFlowGenericInterfaceCounters:
+						gen1 := record.(layers.SFlowGenericInterfaceCounters)
+						gen.IfInOctets += maxuint64(gen1.IfInOctets)
+						gen.IfInUcastPkts += maxuint32(gen1.IfInUcastPkts)
+						gen.IfInMulticastPkts += maxuint32(gen1.IfInMulticastPkts)
+						gen.IfInBroadcastPkts += maxuint32(gen1.IfInBroadcastPkts)
+						gen.IfInDiscards += maxuint32(gen1.IfInDiscards)
+						gen.IfInErrors += maxuint32(gen1.IfInErrors)
+						gen.IfInUnknownProtos += maxuint32(gen1.IfInUnknownProtos)
+						gen.IfOutOctets += maxuint64(gen1.IfOutOctets)
+						gen.IfOutUcastPkts += maxuint32(gen1.IfOutUcastPkts)
+						gen.IfOutMulticastPkts += maxuint32(gen1.IfOutMulticastPkts)
+						gen.IfOutBroadcastPkts += maxuint32(gen1.IfOutBroadcastPkts)
+						gen.IfOutDiscards += maxuint32(gen1.IfOutDiscards)
+						gen.IfOutErrors += maxuint32(gen1.IfOutErrors)
+
+					case layers.SFlowOVSDPCounters:
+						ovsdp1 := record.(layers.SFlowOVSDPCounters)
+						ovsdp.NHit += maxuint32(ovsdp1.NHit)
+						ovsdp.NMissed += maxuint32(ovsdp1.NMissed)
+						ovsdp.NLost += maxuint32(ovsdp1.NLost)
+						ovsdp.NMaskHit += maxuint32(ovsdp1.NMaskHit)
+						ovsdp.NFlows += maxuint32(ovsdp1.NFlows)
+						ovsdp.NMasks += maxuint32(ovsdp1.NMasks)
+
+					case layers.SFlowAppresourcesCounters:
+						app1 := record.(layers.SFlowAppresourcesCounters)
+						app.FdOpen += maxuint32(app1.FdOpen)
+						app.FdMax += maxuint32(app1.FdMax)
+						app.ConnOpen += maxuint32(app1.ConnOpen)
+						app.ConnMax += maxuint32(app1.ConnMax)
+						app.MemUsed += maxuint64(app1.MemUsed)
+						app.MemMax += maxuint64(app1.MemMax)
+
+					case layers.SFlowVLANCounters:
+						vlan1 := record.(layers.SFlowVLANCounters)
+						vlan.Octets += maxuint64(vlan1.Octets)
+						vlan.UcastPkts += maxuint32(vlan1.UcastPkts)
+						vlan.MulticastPkts += maxuint32(vlan1.MulticastPkts)
+						vlan.BroadcastPkts += maxuint32(vlan1.BroadcastPkts)
+						vlan.Discards += maxuint32(vlan1.Discards)
+
+					}
+				}
+			}
+
+			sfa.Graph.Lock()
+			tr := sfa.Graph.StartMetadataTransaction(sfa.Node)
+
+			uint64ToInt64 := func(key uint64) int64 {
```

Comment:
not sure this function is need, you could use `int64(xxxx)` directly line 193 and after